components/email-mgt/org.wso2.carbon.email.mgt/src/main/java/org/wso2/carbon/email/mgt/store/DefaultTemplateManager.java

```diff
@@ -0,0 +1,211 @@
+    public boolean isNotificationTemplateTypeExists(String displayName, String notificationChannel, String tenantDomain)
+            throws NotificationTemplateManagerServerException {
+
+        return templatePersistenceManager.isNotificationTemplateTypeExists(displayName, notificationChannel,
+                tenantDomain) ||
+                inMemoryTemplateManager.isNotificationTemplateTypeExists(displayName, notificationChannel,
+                        tenantDomain);
+    }
+
+    @Override
+    public List<String> listNotificationTemplateTypes(String notificationChannel, String tenantDomain)
+            throws NotificationTemplateManagerServerException {
+
+        List<String> dbBasedTemplateTypes = templatePersistenceManager.listNotificationTemplateTypes
+                (notificationChannel, tenantDomain);
+        List<String> inMemoryTemplateTypes = inMemoryTemplateManager.listNotificationTemplateTypes(notificationChannel,
+                tenantDomain);
+
+        return mergeAndRemoveDuplicates(dbBasedTemplateTypes, inMemoryTemplateTypes);
+    }
+
+    @Override
+    public void deleteNotificationTemplateType(String displayName, String notificationChannel, String tenantDomain)
+            throws NotificationTemplateManagerServerException {
+
+        if (templatePersistenceManager.isNotificationTemplateTypeExists(displayName, notificationChannel,
+                tenantDomain)) {
+            templatePersistenceManager.deleteNotificationTemplateType(displayName, notificationChannel, tenantDomain);
+        }
+    }
+
+    @Override
+    public void addOrUpdateNotificationTemplate(NotificationTemplate notificationTemplate, String applicationUuid,
+                                                String tenantDomain) throws NotificationTemplateManagerServerException {
+
+        templatePersistenceManager.addOrUpdateNotificationTemplate(notificationTemplate, applicationUuid, tenantDomain);
+    }
+
+    @Override
+    public boolean isNotificationTemplateExists(String displayName, String locale, String notificationChannel,
+                                                String applicationUuid, String tenantDomain)
+            throws NotificationTemplateManagerServerException {
+
+        return templatePersistenceManager.isNotificationTemplateExists(displayName, locale, notificationChannel,
+                applicationUuid, tenantDomain) ||
+                inMemoryTemplateManager.isNotificationTemplateExists(displayName, locale, notificationChannel,
+                        applicationUuid, tenantDomain);
+    }
+
+    @Override
+    public NotificationTemplate getNotificationTemplate(String displayName, String locale, String notificationChannel,
+                                                        String applicationUuid, String tenantDomain)
+            throws NotificationTemplateManagerServerException {
+
+        if (templatePersistenceManager.isNotificationTemplateExists(displayName, locale, notificationChannel,
```

Comment:
Lets check the possibility of removing this existence check in a get call.

Reply:
Fixed with https://github.com/wso2-extensions/identity-event-handler-notification/pull/256/commits/8628c42d5f74b1e58a734be7fc641c3f846af46d